webapp/src/app/participant-view/participant-stage-view/exp-chat/exp-chat.component.ts

```diff
@@ -72,12 +73,51 @@ const TIMER_SECONDS = 60; // 1 minute between item pairs for discussions
   styleUrl: './exp-chat.component.scss',
 })
 export class ExpChatComponent implements OnDestroy {
+  // Reload the internal logic dynamically when the stage changes
+  @Input({ required: true })
+  set stage(value: ExpStageChatAboutItems) {
+    this._stage = value;
+
+    this.everyoneReachedTheChat = this.participant.everyoneReachedCurrentStage(this.stage.name);
+
+    // Initialize the current rating to discuss with the first available pair
+    const { id1, id2 } = this.stage.config.ratingsToDiscuss[0];
+    this.currentRatingsToDiscuss = signal({
+      item1: this.stage.config.items[id1],
+      item2: this.stage.config.items[id2],
+    });
+
+    this.unsubscribeMessages = chatMessagesSubscription(this.stage.config.chatId, (m) => {
+      this.messages.set(extendUntilMatch(this.messages(), m.reverse(), 'uid'));
+
+      // Find if new discuss items message have arrived
+      const last = m.find((m) => m.messageType === MessageType.DiscussItemsMessage) as
```

Comment:
Don't use the name name for a list of things as a thing in the list :)  (m in this case).  Use a bit more detailed variable names and it'll be easier to understand.

Reply:
I'm also worried that this will simply find the last pair of items, not tell you if there's a new pair of items to discuss...

Reply:
renamed in 8ef5c53cde2ec9a8f4761e68748efa3b174b8f00

Reply:
I track the current pair being discussed through the `participants_ready_to_end_chat` sync document, which also stores the index of the current pair being discussed in the preconfigured chat list. I should remove one of these 2 ways of computing the last pair being discussed, as they are redundant... If we want to keep the one that scans the last `DIscussItemsMessage` as the single source of truth, we could simply use a `WritableSignal` with lodash's deepEq comparison to make sure that the signal triggers on actual changes